From 77a1fa8cf775bf0bee59aace33d0707bfdd907d6 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 01 Aug 2018 08:11:09 +0300
Subject: [PATCH] T32: update uboot text base

Moved from 0x10000 to 0x100000.
Due to collision of temporary FIT header at SPL while
running some tests for validation (no uboot)
---

diff --git a/board/habanalabs/common/t32/asic.cmm b/board/habanalabs/common/t32/asic.cmm
index 5446871..0205e8c 100644
--- a/board/habanalabs/common/t32/asic.cmm
+++ b/board/habanalabs/common/t32/asic.cmm
@@ -151,9 +151,9 @@
   GOTO waitconnect
 )
 
-; Make sure reset vector for all cores is 0x7ff0010000
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
-D.S axi:0x7FFC441108 %LE %Long 0xf0010000
+; Make sure reset vector for all cores is 0x7ff0100000
+D.S axi:0x7FFC441104 %LE %Long 0xf0100000
+D.S axi:0x7FFC441108 %LE %Long 0xf0100000
 D.S axi:0x7FFC441114 %LE %Long 0x7f
 D.S axi:0x7FFC441118 %LE %Long 0x7f
 
diff --git a/board/habanalabs/common/t32/onix.cmm b/board/habanalabs/common/t32/onix.cmm
index 5713e10..8e1fd9b 100644
--- a/board/habanalabs/common/t32/onix.cmm
+++ b/board/habanalabs/common/t32/onix.cmm
@@ -151,9 +151,9 @@
   GOTO waitconnect
 )
 
-; Make sure reset vector for all cores is 0x7ff0010000
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
-D.S axi:0x7FFC441108 %LE %Long 0xf0010000
+; Make sure reset vector for all cores is 0x7ff0100000
+D.S axi:0x7FFC441104 %LE %Long 0xf0100000
+D.S axi:0x7FFC441108 %LE %Long 0xf0100000
 D.S axi:0x7FFC441114 %LE %Long 0x7f
 D.S axi:0x7FFC441118 %LE %Long 0x7f
 
diff --git a/board/habanalabs/common/t32/pldm.cmm b/board/habanalabs/common/t32/pldm.cmm
index 3cd743a..1474ec8 100644
--- a/board/habanalabs/common/t32/pldm.cmm
+++ b/board/habanalabs/common/t32/pldm.cmm
@@ -151,9 +151,9 @@
   GOTO waitconnect
 )
 
-; Make sure reset vector for all cores is 0x7ff0010000
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
-D.S axi:0x7FFC441108 %LE %Long 0xf0010000
+; Make sure reset vector for all cores is 0x7ff0100000
+D.S axi:0x7FFC441104 %LE %Long 0xf0100000
+D.S axi:0x7FFC441108 %LE %Long 0xf0100000
 D.S axi:0x7FFC441114 %LE %Long 0x7f
 D.S axi:0x7FFC441118 %LE %Long 0x7f
 
diff --git a/board/habanalabs/common/t32/uboot.cmm b/board/habanalabs/common/t32/uboot.cmm
index a7d5560..5c7d7f2 100644
--- a/board/habanalabs/common/t32/uboot.cmm
+++ b/board/habanalabs/common/t32/uboot.cmm
@@ -1,15 +1,11 @@
 ; Make sure reset vector for all cores is 0
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
-D.S axi:0x7FFC441104 %LE %Long 0xf0010000
+D.S axi:0x7FFC441104 %LE %Long 0xf0100000
+D.S axi:0x7FFC441108 %LE %Long 0xf0100000
 D.S axi:0x7FFC441114 %LE %Long 0x7f
-D.S axi:0x7FFC441114 %LE %Long 0x7f
-D.S axi:0x7FFC441114 %LE %Long 0x7f
-D.S axi:0x7FFC441114 %LE %Long 0x7f
+D.S axi:0x7FFC441118 %LE %Long 0x7f
 
 ; Load u-boot
-B::data.LOAD.Binary /lib/firmware/habanalabs/dali/dali-u-boot.bin axi:0x7ff0010000
+B::data.LOAD.Binary /lib/firmware/habanalabs/dali/dali-u-boot.bin axi:0x7ff0100000
 B::data.LOAD.Binary /lib/firmware/habanalabs/dali/dali-u-boot-spl.bin axi:0x7ff0200000
 B::data.LOAD.elf /lib/firmware/habanalabs/dali/u-boot /NoCODE /NoClear
 B::data.LOAD.elf /lib/firmware/habanalabs/dali/u-boot 0x164000 /NoCODE /NoClear
